### PR TITLE
Run pod on all nodes

### PR DESCRIPTION
**Pull request by Arau**
Run collector: One Pod per node when using Node Selectors

When node selectors are defined in a run collector, one Pod is created for each node that satisfies the nodeSelector terms.

In case of needing a Pod to run in one Node specifically, the user must define a nodeSelector that is only fulfilled by one node.

Logs from the Pod are gathered in one action, hence a timeout defined in the run collector applies for the gathering of data from all Pods.

Pods that can't start for entropy reasons, taints, cordons, etc are skipped.

### CODE REVIEW

File: pkg/collect/run.go [hunk 1]
```diff
@@ -23,27 +26,62 @@ func Run(c *Collector, runCollector *troubleshootv1beta2.Run) (map[string][]byte
...
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to run pod")
+	// List of Pods that will be created
+	var pods []corev1.Pod
+
+	if runCollector.NodeSelector != nil {
+		// Get all the nodes where a Pod should be running
+		nodes, err := listNodesInSelectors(ctx, client, runCollector.NodeSelector)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get the list of nodes matching a nodeSelector")
```
Comment (avestuk) [inline pkg/collect/run.go]:
```suggestion
			return nil, fmt.Errorf("failed to get the list of nodes matching a nodeSelector, got err: %w", err)
```

You can also use fmt.Error() to generate an error and using the %w flag will wrap the error. I don't know if what you're doing is an anti pattern, probably Simon is better to ask, but figured I'd point it out.

Comment (Arau) [reply pkg/collect/run.go]: tbh, I used what I saw in that code base. I have no clue what is best, or supposed to do. I'm happy to make that change to have a better code quality :smiley:

Comment (Arau) [reply pkg/collect/run.go]: I applied the error handling to use the fmt.Errorf following the reference https://blog.golang.org/go1.13-errors. 
We are currently not handling the type of the Error in the code that receives the error, however it is good to follow the good practices. 

Thank you!

**Comment by Arau** [comment]
Merging to develop as it already have the main changes of the PR.

**Pull request merged**